types/fsp_test.go

```diff
@@ -16,119 +16,118 @@ package types
 
 import (
 	"strconv"
+	"testing"
 
-	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/require"
 )
 
-var _ = Suite(&FspTest{})
-
-type FspTest struct{}
-
-func (s *FspTest) TestCheckFsp(c *C) {
-	c.Parallel()
+func TestCheckFsp(t *testing.T) {
+	t.Parallel()
 	obtained, err := CheckFsp(int(UnspecifiedFsp))
-	c.Assert(obtained, Equals, DefaultFsp)
-	c.Assert(err, IsNil)
+	require.Equal(t, DefaultFsp, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(-2019)
-	c.Assert(obtained, Equals, DefaultFsp)
-	c.Assert(err, ErrorMatches, "Invalid fsp -2019")
+	require.Equal(t, DefaultFsp, obtained)
+	require.EqualError(t, err, "Invalid fsp -2019")
 
 	obtained, err = CheckFsp(int(MinFsp) - 4294967296)
-	c.Assert(obtained, Equals, DefaultFsp)
-	c.Assert(err, ErrorMatches, "Invalid fsp "+strconv.Itoa(int(MinFsp)-4294967296))
+	require.Equal(t, DefaultFsp, obtained)
+	require.EqualError(t, err, "Invalid fsp "+strconv.Itoa(int(MinFsp)-4294967296))
 
 	// UnspecifiedFsp
 	obtained, err = CheckFsp(-1)
-	c.Assert(obtained, Equals, DefaultFsp)
-	c.Assert(err, IsNil)
+	require.Equal(t, DefaultFsp, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(int(MaxFsp) + 1)
-	c.Assert(obtained, Equals, MaxFsp)
-	c.Assert(err, IsNil)
+	require.Equal(t, MaxFsp, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(int(MaxFsp) + 2019)
-	c.Assert(obtained, Equals, MaxFsp)
-	c.Assert(err, IsNil)
+	require.Equal(t, MaxFsp, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(int(MaxFsp) + 4294967296)
-	c.Assert(obtained, Equals, MaxFsp)
-	c.Assert(err, IsNil)
+	require.Equal(t, MaxFsp, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(int(MaxFsp+MinFsp) / 2)
-	c.Assert(obtained, Equals, (MaxFsp+MinFsp)/2)
-	c.Assert(err, IsNil)
+	require.Equal(t, (MaxFsp+MinFsp)/2, obtained)
+	require.NoError(t, err)
 
 	obtained, err = CheckFsp(5)
-	c.Assert(obtained, Equals, int8(5))
-	c.Assert(err, IsNil)
+	require.Equal(t, int8(5), obtained)
+	require.NoError(t, err)
 }
 
-func (s *FspTest) TestParseFrac(c *C) {
-	c.Parallel()
+func TestParseFrac(t *testing.T) {
+	t.Parallel()
 	obtained, overflow, err := ParseFrac("", 5)
-	c.Assert(obtained, Equals, 0)
-	c.Assert(overflow, Equals, false)
-	c.Assert(err, IsNil)
+	require.Equal(t, 0, obtained)
+	require.Equal(t, false, overflow)
```

Comment:
Thanks for your contribution. Almost OK, using `require.False(t, overflow)`  instead of `require.Equal(t, false, overflow)` will be better.

Reply:
> Thanks for your contribution. Almost OK, using `require.False(t, overflow)` instead of `require.Equal(t, false, overflow)` will be better.

Done.